app/code/Magento/Catalog/view/adminhtml/web/js/product/form/fieldset/custom-options.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'Magento_Ui/js/form/components/fieldset',
+    'uiRegistry'
+], function (Component, uiRegistry) {
+    'use strict';
+
+    return Component.extend({
+        /**
+         * Handler of the "opened" property changes.
+         *
+         * @param {Boolean} isOpened
+         */
+        onVisibilityChange: function (isOpened) {
+            this._super(isOpened);
+
+            // Once the Custom Options tab is opened change "affect_product_custom_options" value to true
+            // in order custom options be processed on backend.
+            if (this.opened()) {
+                uiRegistry.get('product_form.product_form.custom_options.affect_product_custom_options').value(1);
```

Comment:
Would be good to use async way for retrieving this component, like described in https://github.com/magento/magento2/blob/9544fb243d5848a497d4ea7b88e08609376ac39e/app/code/Magento/Ui/view/base/web/js/lib/registry/registry.js#L342-L346

```suggestion
                uiRegistry.get('product_form.product_form.custom_options.affect_product_custom_options', function(component) {
                    component.value(1);
                });
```